### PR TITLE
Axis updates

### PR DESCRIPTION
**Pull request by martinjrobins**
commits for #3. Plus I removed the clang-tidy `cppcoreguidelines-pro-bounds-constant-array-index` test

### CODE REVIEW

File: src/Backend.hpp [hunk 1]
```diff
@@ -34,8 +34,95 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+#include <cmath>
+#include <sstream>
+
+#include "Vector.hpp"
+
 namespace trase {
 
+// pi constant
+const float pi =
+    3.141592653589793238462643383279502884197169399375105820974944592307816406286;
```
Comment (fcooper8472) [inline src/Backend.hpp]: I think there's enough precision here ;)

File: src/AxisDraw.hpp [hunk 2]
```diff
@@ -139,6 +138,35 @@ template <typename Backend> void Axis::draw_common(Backend &backend) {
   backend.stroke_color(RGBA(255, 255, 255, 255));
   backend.stroke_width(lw / 2);
   backend.stroke();
+
+  // axis title
+  if (!m_title.empty()) {
+    backend.text_align(ALIGN_CENTER | ALIGN_BOTTOM);
+    backend.text(vfloat2_t(0.5 * (m_pixels.bmax[0] + m_pixels.bmin[0]),
+                           m_pixels.bmin[1] -
+                               0.05 * (m_pixels.bmax[1] - m_pixels.bmin[1])),
```
Comment (fcooper8472) [inline src/AxisDraw.hpp]: Is appveyor's build complaining that these are `0.5` instead of `0.5f`?

File: src/Backend.hpp (hunk 1 above)
Comment (martinjrobins) [reply src/Backend.hpp]: :) I was wondering if that would even fit in a float!

Comment (fcooper8472) [reply src/Backend.hpp]: Also, that can presumably be `constexpr` not just `const`?

**Comment by martinjrobins** [comment]
Yup, just saw those. I've tried to fix them and added some more axis tests while I was there. see how windows feels about it now....

**Pull request merged**